### PR TITLE
Responsive Classic CSS - Mostly duplicate line removal

### PR DESCRIPTION
**Pull request by TMCSherpa**
Mostly duplicate line removal PR split part from #6747

### CODE REVIEW

**Comment by TMCSherpa** [comment]
Just so everyone is on the same page, here is a crappy script fusion of what I did
START
#!/bin/bash

cat ./css/stylesheet.css |grep "\S" |paste -s -d ''|tr -s '[:space:]'|sed 's/,\s/,/g '|sed 's/\s,/,/g'|sed 's/\s{/{/g'|sed 's/}\s/}/g'|sed 's/}/}\n/g'|sed 's/\/\*.*\*\///'> ./stylesheet_copy.txt
cat stylesheet_copy.txt| sed 's#{#,\n{#g' > stylesheet_split.txt
cat stylesheet_split.txt |grep { > stylesheet_properties.txt
cat stylesheet_split.txt |grep -v { > stylesheet_elements.txt
rm stylesheet_copy.txt
rm stylesheet_split.txt

x=1
while IFS= read -r line; do
    sed -n "$x p" stylesheet_elements.txt|sed "s&,&$line\n&g"|grep "\S" >> stylesheet_per_line.txt
    x=$(( $x + 1 ))
done < stylesheet_properties.txt
rm stylesheet_properties.txt
rm stylesheet_elements.txt

cat stylesheet_per_line.txt |sed 's/{.*}//'|sort -u > stylesheet_diff_list.txt

x=1
while IFS= read -r line; do
    echo "working on "$line
    echo -e "\t" $line >> stylesheet_dup.txt
    cat stylesheet_per_line.txt |grep ^"$line"\{ >> stylesheet_dup.txt
    echo >> stylesheet_dup.txt
    x=$(( $x + 1 ))
done < stylesheet_diff_list.txt
rm stylesheet_diff_list.txt
rm stylesheet_per_line.txt

cat ./css/stylesheet_colors.css |grep "\S" |paste -s -d ''|tr -s '[:space:]'|sed 's/,\s/,/g '|sed 's/\s,/,/g'|sed 's/\s{/{/g'|sed 's/}\s/}/g'|sed 's/}/}\n/g'|sed 's/\/\*.*\*\///'> ./stylesheet_colors_copy.txt
cat stylesheet_colors_copy.txt| sed 's#{#,\n{#g' > stylesheet_colors_split.txt
cat stylesheet_colors_split.txt |grep { > stylesheet_colors_properties.txt
cat stylesheet_colors_split.txt |grep -v { > stylesheet_colors_elements.txt
rm stylesheet_colors_copy.txt
rm stylesheet_colors_split.txt

x=1
while IFS= read -r line; do
    sed -n "$x p" stylesheet_colors_elements.txt|sed "s&,&$line\n&g"|grep "\S" >> stylesheet_colors_per_line.txt
    x=$(( $x + 1 ))
done < stylesheet_colors_properties.txt
rm stylesheet_colors_properties.txt
rm stylesheet_colors_elements.txt

cat stylesheet_colors_per_line.txt |sed 's/{.*}//'|sort -u > stylesheet_colors_diff_list.txt

x=1
while IFS= read -r line; do
    echo "working on "$line
    echo -e "\t" $line >> stylesheet_colors_dup.txt
    cat stylesheet_colors_per_line.txt |grep ^"$line"\{ >> stylesheet_colors_dup.txt
    echo >> stylesheet_colors_dup.txt
    x=$(( $x + 1 ))
done < stylesheet_colors_diff_list.txt
rm stylesheet_colors_diff_list.txt
rm stylesheet_colors_per_line.txt
END

run it in the responsive_clasic folder and look for blocks of lines

from stylesheet
	 #mainWrapper
#mainWrapper{text-align:left;width:100%;vertical-align:top;}
#mainWrapper{margin:0;padding:0;}
#mainWrapper{margin:auto;} <- overrides margin:0
#mainWrapper{} <- plenty of #mainWrapper lines, does this need to be here?

from stylesheet_colors
	 #mainWrapper
#mainWrapper{background:#fff;} <- these are both on line 36 for some reason
#mainWrapper{background:#fff;}
#mainWrapper{background:#f4f4f4;} <- line 37 overrides (both) #fff

**Comment by drbyte** [comment]
There are a lot of changes here. Super hard to review.

Why are we doing this?

**Comment by TMCSherpa** [comment]
For normal people? General cleanup. Combining things like margin, margin-top, margin-bottom so if you're trying to make a template everything is in the same place.
For non-normal people? I have a couple scripts for comparing the responsive_default files with the templates from Picaflor (Dover, Winchester etc) to see what's changed. It's about 500 selectors gained/lost/renamed for Dover_Fine for instance.
Having clean files on both sides should make mashing them together easier.

**Comment by TMCSherpa** [comment]
Okay, here is what was removed

Duplicates

#cartBoxListWrapper{margin:0;padding:0;}
#cartBoxListWrapper{margin:0;padding:0;}

#navEZPagesTOC ul li a{padding:0 0.5em;margin:0;text-decoration:underline;}
#navEZPagesTOC ul li a{text-decoration:underline;}

#navMainSearch input[type=submit]{-moz-box-sizing: border-box; -webkit-box-sizing: border-box; box-sizing: border-box;display:inline-block;margin:0;height:32px;}
#navMainSearch input[type=submit]{-moz-box-sizing: border-box; -webkit-box-sizing: border-box; box-sizing: border-box;display:inline-block;margin:0;height:32px;}

If there is a {} but it has something elsewhere it's gone
#headerWrapper{}
#headerWrapper{margin:0;padding:0;}
#mainWrapper{}
#mainWrapper{margin:0;padding:0;}
.messageStackCaution{}
.messageStackCaution{line-height:1.8em;padding:20px;}
.messageStackError{}
.messageStackError{line-height:1.8em;padding:20px;}
.messageStackSuccess{}
.messageStackSuccess{line-height:1.8em;padding:20px;}
.messageStackWarning{}
.messageStackWarning{line-height:1.8em;padding:20px;}
.productPriceDiscount{}
.productPriceDiscount{margin:10px 0 5px 0;display:block;font-size:90%;}
.productSpecialPrice{}
.productSpecialPrice{font-size:120%;}
.productSpecialPriceSale{}
.productSpecialPriceSale{text-decoration:line-through;}
.tableHeading{}
.tableHeading{height:2.2em;}

Stuff like this where it's the only reference stays
/*top ezpages links*/
#navEZPagesTop {margin:0;width:100%;}
#navEZPagesTop ul {font-size:0;text-align:center;overflow:hidden;}
#navEZPagesTop li {font-size:0.75rem;display: inline-block;margin-top:0;}
#navEZPagesTop ul li:last-child{}
#navEZPagesTop li a {text-decoration: underline;padding: 1.2em 1.0em;display: block;}
#navEZPagesTop li a:hover {}

The stuff moved to stylesheet_colors
Everything that is a background was in colors already so was removed
All the borders removed from stylesheet and added to /*bof border colors*/ in stylesheet_colors
h3.leftBoxHeading a{color: #ffffff;font-weight:normal;text-decoration:underline;}
h3.rightBoxHeading a{color: #ffffff;font-weight:normal;text-decoration:underline;}
#popupAdditionalImage{background:none;}
#popupAtrribsQuantityPricesHelp{background:none;}
#popupCouponHelp{background:none;}
#popupCVVHelp{background:none;}
#popupImage{background:none;}
#popupSearchHelp{background:none;}
#popupShippingEstimator{background:none;}
#docGeneralDisplay #pinfo-right{background:none;}
fieldset fieldset legend{text-align:left;padding:0;margin:0;position:relative;left:20px;font-size:120%;display:table;background:none;}
#infoShoppingCart{background:none;}
input[type=file]{border:none;width:auto;height:auto;}
input[type=image]{border:none;width:auto;height:auto;}
.last{border:none !important;}
.pagination li:first-child{border-left:0;}
#productQuantityDiscounts table{margin:0 auto 20px auto;border:none !important;border-spacing:0 !important;border-collapse:separate !important;}
#productQuantityDiscounts table tr:first-child td{padding:5px 10px;border:none !important;}
#productQuantityDiscounts td:last-child{border:none;}
#productQuantityDiscounts td{padding:10px;border:none;}
select{border:none;padding-left:5px;width:51%;}
#shippingEstimatorContent table{border:none !important;margin-top:30px;}
#shippingEstimatorContent td{border:none !important;}
#shippingEstimatorContent th{border:none !important;}
.tabTable{width:100%;border:none;padding:0;margin:0; border-spacing:0;border-collapse:collapse;}
#accountDefault #sendSpendWrapper{width:100%;border:none;}
a img{border:none;}
fieldset{border:0;padding:0.01em 0 0 0;padding-bottom:20px;padding-top:20px;margin:0;min-width:0;margin-bottom:20px;}
HR{height: 1px;margin-top:0.5em;border:none;}
.blue{}
.blue:hover{}

The sticky wicket
These are NOT in line order so they probably don't line up with the dif changes
Changing width, padding or margin settings combined

.tabTable{margin-bottom:20px;} <-line 315
.tabTable{width:100%;border:none;padding:0;margin:0; border-spacing:0;border-collapse:collapse;} <-line 486

#accountDefault #sendSpendWrapper h2{margin-top:-20px;clear:both;}
#accountDefault #sendSpendWrapper h2{padding:10px;font-weight:normal;margin-bottom:0;text-align:center;}

#accountLinksWrapper{margin:0;width:47%;}
#accountLinksWrapper{width:100%;}

#addressBookDefault .instructions{padding:20px 20px 20px 30px;margin-top:0;}
#addressBookDefault .instructions{padding-bottom:40px;}

#bannerSix{font-weight:normal;padding:1.0em 0;margin:0;}
#bannerSix{padding:0.8em;}

.cartBoxTotal{margin:0;padding: 0.2em 0;}
.cartBoxTotal{padding:10px !important;}

.cartTotalDisplay{padding: 0.5em 0;}
.cartTotalDisplay{text-align:right;padding-right:0.2em;}

.cartUnitDisplay{padding: 0.5em 0;}
.cartUnitDisplay{text-align:right;padding-right:0.2em;}

#contentMainWrapper{margin:0;padding:0;}
#contentMainWrapper{padding-bottom:30px;}

#featuredProductsListingBottomNumber{margin-left:-71%;}
#featuredProductsListingBottomNumber{width:25%;margin-left:-74%;margin-top:0;}

#featuredProductsListingTopNumber{margin-left:-71%;}
#featuredProductsListingTopNumber{width:25%;margin-left:-74%;margin-top:0;}

.floatingBox{margin:0;width:47%;}
.floatingBox{width:46%;}

h4.optionName{line-height:1.5em;padding:0.2em;}
h4.optionName{margin:1% 30px 0 30px;padding:0;}

#indexProductListCatDescription{margin:0.5em;}
#indexProductListCatDescription{margin-left:5%;float:left;}

#mainWrapper{margin:0;padding:0;}
#mainWrapper{margin:auto;}

#navEZPagesTOCWrapper{float:right;padding:0 20px 20px 20px;margin:0 0 20px 20px;}
#navEZPagesTOCWrapper{margin:0;width:47%;}

#newProductsDefaultListingBottomNumber{margin-left:-72.5%;}
#newProductsDefaultListingBottomNumber{width:25%;margin-left:-74%;margin-top:0;}

#newProductsDefaultListingTopNumber{margin-left:-72.5%;}
#newProductsDefaultListingTopNumber{width:25%;margin-left:-74%;margin-top:0;}

#productMainImage{float:none;margin:0;}
#productMainImage{margin:0 1em 1em 0;}

#reviewWriteMainImage{float:none;margin:0;}
#reviewWriteMainImage{margin:0 1em 1em 0;}

#sendSpendWrapper h2{margin-top:-20px;}
#sendSpendWrapper h2{padding:10px;font-weight:normal;margin-bottom:0;text-align:center;}

Everything in FireFox's inspector looks correct after all the changes

File: includes/templates/responsive_classic/css/stylesheet.css [hunk 1]
```diff
@@ -57,19 +54,19 @@ HR {height: 1px;margin-top:0.5em;border:none;}
...
 body:not(:-moz-handler-blocked) fieldset{display:table-cell;}
 fieldset > fieldset{}
-fieldset fieldset legend{text-align:left;padding:0;margin:0;position:relative;left:20px;font-size:120%;display:table;background:none;}
+fieldset fieldset legend{text-align:left;padding:0;margin:0;position:relative;left:20px;font-size:120%;display:table;}
 fieldset p{margin-left:20px;}
 fieldset label, fieldset .inputLabel, fieldset h3, fieldset address, #advSearchDefault input, fieldset .buttonRow.back, fieldset select, fieldset input[type=radio], fieldset input[type=checkbox], fieldset div{margin-left:20px;}
 fieldset div, fieldset .buttonRow.forward{margin-right:20px;}
 option{font-size:110%;}
-LABEL, h4.optionName {line-height:1.5em;padding:0.2em;}
+LABEL {line-height:1.5em;padding:0.2em;}
```
Comment (drbyte) [inline includes/templates/responsive_classic/css/stylesheet.css:69]: I'm not seeing where `h4.optionName` (or `h4` or `.optionName`) is ever assigned `line-height` property elsewhere in any of the RC CSS.

File: includes/templates/responsive_classic/css/stylesheet.css [hunk 2]
```diff
@@ -20,15 +20,15 @@ a:active {}
...
 P, ADDRESS {padding:0.5em;}
 ADDRESS {font-style:normal;}
-HR {height: 1px;margin-top:0.5em;border:none;}
+HR {height: 1px;margin-top:0.5em;}
 #logo, .centerBoxContents, .specialsListBoxContents, .categoryListBoxContents, .centerBoxContentsAlsoPurch, .attribImg {float:left;}
-.centeredContent, TH, #cartEmptyText, #cartBoxGVButton, #cartBoxEmpty, #cartBoxVoucherBalance, #navCatTabsWrapper, #navEZPageNextPrev, #bannerOne, #bannerTwo,  #bannerThree,  #bannerFour,  #bannerFive,  #bannerSix, #siteinfoLegal, #siteinfoCredits, #siteinfoStatus, #siteinfoIP, .center, .cartRemoveItemDisplay, .cartQuantityUpdate, .cartQuantity, .cartTotalsDisplay, #cartBoxGVBalance, .leftBoxHeading, .centerBoxHeading,.rightBoxHeading, .productListing-data, .accountQuantityDisplay, .ratingRow, LABEL#textAreaReviews, #productMainImage, #reviewsInfoDefaultProductImage, #productReviewsDefaultProductImage, #reviewWriteMainImage, .centerBoxContents, .specialsListBoxContents, .categoryListBoxContents, .additionalImages, .centerBoxContentsSpecials, .centerBoxContentsFeaturedCategories, .centerBoxContentsAlsoPurch, .centerBoxContentsFeatured, .centerBoxContentsNew, .gvBal, .attribImg {text-align:center;}
```
Comment (drbyte) [inline includes/templates/responsive_classic/css/stylesheet.css:30]: `.accountQuantityDisplay` is not assigned to `text-align: center` anywhere else.

Comment (drbyte) [reply includes/templates/responsive_classic/css/stylesheet.css:30]: Oops. Yes it is.

File: includes/templates/responsive_classic/css/stylesheet.css (hunk 1 above)
Comment (drbyte) [reply includes/templates/responsive_classic/css/stylesheet.css:69]: Argh. Yes, found it.

File: includes/templates/responsive_classic/css/stylesheet.css [hunk 3]
```diff
@@ -259,14 +249,14 @@ TR.tableHeading {}
...
+#sendSpendWrapper h2{padding:10px;font-weight:normal;margin-top:-20px;margin-bottom:0;text-align:center;}
 
 /*bof images*/
-#productMainImage, #reviewsInfoDefaultProductImage, #productReviewsDefaultProductImage, #reviewWriteMainImage {margin:0 1em 1em 0;}
+#reviewsInfoDefaultProductImage, #productReviewsDefaultProductImage {margin:0 1em 1em 0;}
 .cartImage {margin:0.5em 1em;}
 
 /*bof center boxes*/
-.centerBoxContents, .specialsListBoxContents, .categoryListBoxContents, .additionalImages, .centerBoxContentsSpecials, .centerBoxContentsFeaturedCategories, .centerBoxContentsAlsoPurch, .centerBoxContentsFeatured, .centerBoxContentsNew {margin:1em 0;}
+.centerBoxContents, .specialsListBoxContents, .categoryListBoxContents, .centerBoxContentsSpecials, .centerBoxContentsFeaturedCategories, .centerBoxContentsAlsoPurch, .centerBoxContentsFeatured, .centerBoxContentsNew {margin:1em 0;}
```
Comment (drbyte) [inline includes/templates/responsive_classic/css/stylesheet.css:259]: Where is the `. additionalImages` assigned a margin now?

File: includes/templates/responsive_classic/css/stylesheet.css [hunk 4]
```diff
@@ -259,14 +249,14 @@ TR.tableHeading {}
 #checkoutOrderTotals div, #csNotifications .buttonRow{margin-right:20px;}
 #checkoutPayment label, #csNotifications input{margin-left:20px;}
 div#checkoutShippingContentChoose{margin-bottom:20px;}
-#sendSpendWrapper h2{margin-top:-20px;}
+#sendSpendWrapper h2{padding:10px;font-weight:normal;margin-top:-20px;margin-bottom:0;text-align:center;}
 
 /*bof images*/
-#productMainImage, #reviewsInfoDefaultProductImage, #productReviewsDefaultProductImage, #reviewWriteMainImage {margin:0 1em 1em 0;}
+#reviewsInfoDefaultProductImage, #productReviewsDefaultProductImage {margin:0 1em 1em 0;}
```
Comment (drbyte) [inline includes/templates/responsive_classic/css/stylesheet.css:255]: Your comments say:
```
#productMainImage{float:none;margin:0;}
#productMainImage{margin:0 1em 1em 0;}

#reviewWriteMainImage{float:none;margin:0;}
#reviewWriteMainImage{margin:0 1em 1em 0;}
```

... but I'm only seeing the removal here, not the relocation or combination.

File: includes/templates/responsive_classic/css/stylesheet.css (hunk 3 above)
Comment (drbyte) [reply includes/templates/responsive_classic/css/stylesheet.css:259]: Okay, it was part of existing content, not changed content.

File: includes/templates/responsive_classic/css/stylesheet.css (hunk 4 above)
Comment (drbyte) [reply includes/templates/responsive_classic/css/stylesheet.css:255]: Found it. Existing content, not part of diff "changes".

File: includes/templates/responsive_classic/css/stylesheet_colors.css [hunk 5]
```diff
@@ -9,32 +9,29 @@
...
 .cartNewItem {color:#DB3A00;}
 #orderhistoryContent ul li a i.fa,
 .productSpecialPrice, .productSalePrice, .productSpecialPriceSale, .productPriceDiscount {color:#900404;}
 .categoryListBoxContents a:hover, .categoryListBoxContents:hover a{color:#364fb5;}
 .list-more{color:#fff !important;}
 
 /* Added by dbltoe for more accent on center page items.  Comment out any not wanted */
-.categoryListBoxContents:hover, centerBoxContentsListing:hover, .centerBoxContents:hover, .centerBoxContentsNew:hover, .centerBoxContentsFeatured:hover, .centerBoxContentsSpecials:hover, .specialsListBoxContents:hover, .centerBoxContentsFeaturedCategories:hover {border-radius: 5px;background:#e2e3e4;box-shadow: 21px 21px 41px #bdbdbd, -21px -21px 41px #d9d9d9;}
+centerBoxContentsListing:hover, .centerBoxContents:hover, {border-radius: 5px;background:#e2e3e4;box-shadow: 21px 21px 41px #bdbdbd, -21px -21px 41px #d9d9d9;}
```
Comment (drbyte) [inline includes/templates/responsive_classic/css/stylesheet_colors.css:29]: Missing `.` prefix at beginning of this line.
```suggestion
.centerBoxContentsListing:hover, .centerBoxContents:hover, {border-radius: 5px;background:#e2e3e4;box-shadow: 21px 21px 41px #bdbdbd, -21px -21px 41px #d9d9d9;}
```

Comment (drbyte) [reply includes/templates/responsive_classic/css/stylesheet_colors.css:29]: Hmmm ... The before/after run had the `.` in  the "before". But now I can't figure out why.

Regardless, it "should" be there, as without the `.` it's pointless.

**Pull request merged**